backend/api/contacts/src/main/java/co/airy/core/contacts/dto/Contact.java

```diff
@@ -159,6 +167,29 @@ public Address merge(Address address) {
             .collect(Collectors.toList());
     }
 
+    @JsonIgnore
+    public Contact merge(Contact c) {
+        return this.toBuilder()
+                .metadata(Optional.ofNullable(this.getMetadata()).orElse(c.getMetadata()))
+                .address(this.getAddress().merge(c.getAddress()))
+                // Aggregate conversations
+                .conversations(Stream.concat(
+                            this.getConversations().entrySet().stream(),
+                            c.getConversations().entrySet().stream())
```

Comment:
For now I am not filtering common conversations because if I am not mistaken they should not have common conversations